Leave game event added.

LudvigStridsberg opened the pull request:
Leave game event added and also put in events/index.js. What do you guys think about it? Is something missing?

LudvigStridsberg committed:
Leave game event added.

Leave game event added and also put in events/index.js.

trumtomte commented on events/gameLeave.js:
```diff
@@ -0,0 +1,29 @@
+    const {gameID} = payload;
+
+    // Get the game
+    const game = context.getGameState({gameID});
```
For now, lets skip using `gameId`, since if a player is leaving a game they should already be connected to one. Try just calling `context.getGameState()` instead. And to verify, check that `game !== null`.

trumtomte commented on events/gameLeave.js:
```diff
@@ -0,0 +1,29 @@
+    const playerID = context.id();
+
+    // Filter away all players that aren't the one who is leaving
+    let otherPlayers = game.players.filter(function(player) {
```
Do note that `game.players` is an object, not an array. To delete from an array you could use the `delete` keyword, that is: `delete game.players[playerId]`.

trumtomte commented on events/gameLeave.js:
```diff
@@ -0,0 +1,29 @@
+    game.players = otherPlayers;
+
+    // Uptade the game with the new player-array
+    context.updateGameState(game, gameID);
```
Since the player is connected to a game we don't need to specify `gameId`.

trumtomte commented on events/gameLeave.js:
```diff
@@ -0,0 +1,29 @@
+
+    // Uptade the game with the new player-array
+    context.updateGameState(game, gameID);
+
```
We do however need to update the player state as well. That would be:
``` js
const player = context.getPlayerState();
player.gameId = null;
context.updatePlayerState(player);
```

trumtomte reviewed (changes requested):
A really good start, added some notes that should be easily fixed.

LudvigStridsberg committed:
Made requested changes

Updated the file so that Sebastian's requested changes were implemented. The file no longer gets the game's id, it checks so that game isn't null, it uses delete instead of filter and it updates the player's state as well.

Co-Authored-By: Sebastian Bengtegård <sebastianbengtegard@gmail.com>

trumtomte reviewed (approved):
Nice! 👍🏻

Commit:
Merge branch 'main' into Event-Leave-Game
